### PR TITLE
rtc: Add data models and parser for AECs

### PR DESCRIPTION
**Pull request by jtrh**
- Add data model for the SII's "Archivo Electrónico de Cesión" (AEC) in XML format.
- Add data model for the "cesiones" contained in an AEC.
- Add parser for AECs in XML format.
- Add tests.

### CODE REVIEW

**Review by yaselc — COMMENTED** [review]
Wow... 🤯 
Just add that you should include the "Identificador de Envio" field, from the RPETC email:
`RPETC email: attachment / 'Identificador de Envio'`
This is a key piece of information to check the status of an AEC submission and can only be obtained from the RPETC email or submitting the AEC.

**Comment by jtrh** [comment]
> Just add that you should include the "Identificador de Envio" field, from the RPETC email:
> `RPETC email: attachment / 'Identificador de Envio'`
> This is a key piece of information to check the status of an AEC submission and can only be obtained from the RPETC email or submitting the AEC.

@yaselc  Can I extract the "identificador de envío" from the contents of an **AEC XML file**?

The scope of this pull request is modelling and parsing an AEC in XML format, not other related entities such as the RPETC email, unless those entities are embedded in the AEC XML file.

**Comment by jtrh** [comment]
> The scope of this pull request is modelling and parsing an AEC in XML format, not other related entities such as the RPETC email, unless those entities are embedded in the AEC XML file.

Of course, that does not exclude the possibility of creating a separate data model for the RPETC Email Message in a future pull request.

File: cl_sii/rtc/data_models_aec.py [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+    @pydantic.validator('cesiones')
+    def validate_cesiones_seq_order(cls, v: object) -> object:
+        if isinstance(v, Sequence):
+            for idx, cesion in enumerate(v, start=1):
+                if cesion.seq != idx:
+                    raise ValueError("items must be ordered according to their 'seq'")
+        return v
+
+    @pydantic.validator('cesiones')
+    def validate_cesiones_monto_cesion_must_not_increase(cls, v: object) -> object:
```
[the remark this answers is not in the document]
Comment (jtrh) [reply cl_sii/rtc/data_models_aec.py:683]: > [...] validate that `cesiones_monto_cesion` never exceeds the `dte_monto_total`

I've got that covered. `CesionAecXml.validate_monto_cesion_does_not_exceed_dte_monto_total` performs that check at the individual _cesión_ level, and `AecXml.validate_dte_matches_cesiones_dtes` checks that all _cesiones_ have the same _DTE_ information, so no _cesión_ can ever exceed the _DTE's_ total amount.

**Comment by yaselc** [comment]
> @yaselc Can I extract the "identificador de envío" from the contents of an **AEC XML file**?
> 
> The scope of this pull request is modelling and parsing an AEC in XML format, not other related entities such as the RPETC email, unless those entities are embedded in the AEC XML file.

Makes sense, but I'm thinking here of the use case that FP generates the AEC and submits it to the SII, the response is the "Identificador de Envio" and FP will want to store this AEC together with the "Identificador de Envio" to later check its status.
Even if the responsibility for generating and submitting the AEC were assigned to FD, it would still need to store it.

**Comment by glarrain** [comment]
> Of course, that does not exclude the possibility of creating a separate data model for the RPETC Email Message in a future pull request.

I agree

**Comment by jtrh** [comment]
> > @yaselc Can I extract the "identificador de envío" from the contents of an **AEC XML file**?
> > The scope of this pull request is modelling and parsing an AEC in XML format, not other related entities such as the RPETC email, unless those entities are embedded in the AEC XML file.
> 
> Makes sense, but I'm thinking here of the use case that FP generates the AEC and submits it to the SII, the response is the "Identificador de Envio" and FP will want to store this AEC together with the "Identificador de Envio" to later check its status.
> Even if the responsibility for generating and submitting the AEC were assigned to FD, it would still need to store it.

@yaselc: We can store the submission identifier in an optional field of the AEC Django model of FD CL Data.

**Review by yaselc — APPROVED** [review]

**Pull request merged**